### PR TITLE
Move type definitions to separate package

### PR DESCRIPTION
**Pull request by dagurval**
Moves type definitions to a new package @polkadot/joy-types. This allows
for importing them into other projects without dragging in loads of
unrelated dependencies.

### CODE REVIEW

File: packages/joy-types/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "@polkadot/joy-types",
+  "version": "0.1.0",
+  "description": "Joystream types",
```
Comment (siman) [inline packages/joy-types/package.json]:
```suggestion
  "description": "Joystream types for Substrate modules",
```

**Comment by jfinkhaeuser** [comment]
Is there a particular reason for the name? I mean, nothing much in this package relates to polkadot. I mean, shouldn't it just be `joy-types` ?

**Comment by dagurval** [comment]
> Is there a particular reason for the name? I mean, nothing much in this package relates to polkadot. I mean, shouldn't it just be `joy-types` ?

Just for consistency, every other package is @polkadot/joy-<something>. I don't mind using another name.

**Comment by jfinkhaeuser** [comment]
Sure. I guess it's more that these `@polkadot` names relate to specifying e.g. `@polkadot/types` as a dependency. So `@polkadot` is a namespace for, well, polkadot packages.

When it comes to packages that are exclusive to the app, I'm thinking the naming barely matters. But this is something that could be published independently. So that's where I'm not sure about the name.

It's a minor thing to be sure, but I know @siman and he tends to have thoughts on this stuff - so I wanted to raise it!

**Comment by siman** [comment]
> > Is there a particular reason for the name? I mean, nothing much in this package relates to polkadot. I mean, shouldn't it just be `joy-types` ?
> 
> Just for consistency, every other package is @polkadot/joy-<something>. I don't mind using another name.

@dagurval try to rename `@polkadot/joy-types` to `@joystream/joy-types` in all files that are affected.

I cannot remember why I left all our modules prefixed with `@polkadot` instead of `@joystream`... maybe I kept `@polkadot` because we forked Polkadot Apps and were not sure that we would stick to it so I didn't make lots of changes into the app structure.

**Comment by dagurval** [comment]
files renamed, package renamed and description fixed

**Comment by dagurval** [comment]
Revisiting the storage module, I just realized this somewhat duplicates effort done there: https://github.com/Joystream/storage-node-joystream/tree/master/lib/joystream/substrate/types

**Comment by dagurval** [comment]
renamed `@joystream/joy-types` to `@joystream/types` as suggested by @siman

**Pull request merged**